Orange/data/variable.py

```diff
@@ -700,13 +709,16 @@ def to_val(self, s):
         if not isinstance(s, str):
             raise TypeError('Cannot convert {} to value of "{}"'.format(
                 type(s).__name__, self.name))
-        return self.values.index(s)
+        if s not in self._value_index:
+            raise ValueError(f"Value {s} does not exist")
+        return self._value_index[s]
 
     def add_value(self, s):
         """ Add a value `s` to the list of values.
         """
         if not isinstance(s, str):
             raise TypeError("values of DiscreteVariables must be strings")
+        self._value_index[s] = len(self.values)
```

Comment:
Since this is a public method, I'd add a check `if s not in self._value_index`, before inserting a new value.

Reply:
Same as above. I agree, but this is from #4422, which is already merged.

Reply:
I now implemented this in #4450.

Plus fixing some lint issues for a good measure.